docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
+>
+> The `osbuilder` tool provides
+> [configurations for various common Linux distributions](../../tools/osbuilder/rootfs-builder)
+> which can be built into either initrd or rootfs guest images.
 
 #### Root filesystem image
 
-The default packaged root filesystem image, sometimes referred to as the "mini O/S", is a
-highly optimized container bootstrap system based on [Clear Linux](https://clearlinux.org/). It provides an extremely minimal environment and
-has a highly optimized boot path.
+The default packaged root filesystem image ("rootfs"), sometimes
+referred to as the "mini O/S", is a highly optimized container
+bootstrap system based on [Clear Linux](https://clearlinux.org). It
+provides an extremely minimal environment and has a highly optimized
+boot path.
+
+The only services running in the context of the mini O/S are:
+
+| Process | Environment | systemd service? | Notes |
+|-|-|-|-|
+| systemd | VM root | n/a | The init daemon, running as PID 1 |
+| [Agent](#agent) | VM root | yes | Runs as a systemd service |
+| `chronyd` | VM root | yes | Used to synchronise the time with the host |
+| container workload | VM container | no | Managed by the agent |
+
+See also the [process overview](#process-overview).
 
-The only services running in the context of the mini O/S are the init daemon
-(`systemd`) and the [Agent](#agent). The real workload the user wishes to run
-is created using libcontainer, creating a container in the same manner that is done
-by `runc`.
+> **Note:** The container workload is running inside a full container
+> environment which itself is running within a VM environment.
 
-For example, when `ctr run -ti ubuntu date` is run:
+For example, when the user runs `ctr run -ti ubuntu date`:
 
 - The hypervisor will boot the mini-OS image using the guest kernel.
 - `systemd`, running inside the mini-OS context, will launch the `kata-agent` in
   the same context.
 - The agent will create a new confined context to run the specified command in
   (`date` in this example).
 - The agent will then execute the command (`date` in this example) inside this
-  new context, first setting the root filesystem to the expected Ubuntu\* root
+  new context, first setting the root filesystem to the expected Ubuntu root
   filesystem.
 
 #### Initrd image
 
-A compressed `cpio(1)` archive, created from a rootfs which is loaded into memory and used as part of the Linux startup process. During startup, the kernel unpacks it into a special instance of a `tmpfs` that becomes the initial root filesystem.
+The initrd image is a compressed `cpio(1)` archive, created from a
+rootfs which is loaded into memory and used as part of the Linux
+startup process. During startup, the kernel unpacks it into a special
+instance of a `tmpfs` mount that becomes the initial root filesystem.
+
+The only service running in the context of the initrd is the
+[Agent](#agent) as the init daemon. The real workload the user wishes
```

Comment:
We still support systemd as init, as that's what Red Hat is using on their product.